### PR TITLE
Ruby 3.1 support

### PR DESCRIPTION
**Pull request by KitaitiMakoto**
@fbernier Can you consider?

### CODE REVIEW

File: tomlrb.gemspec [hunk 1]
```diff
@@ -17,4 +17,6 @@ Gem::Specification.new do |spec|
   spec.files         = %w{LICENSE.txt} + Dir.glob("lib/**/*", File::FNM_DOTMATCH).reject { |f| File.directory?(f) }
   spec.require_paths = ["lib"]
   spec.required_ruby_version = '>= 2.0'
+
+  spec.add_development_dependency "psych", "~> 4"
```
Comment (fbernier) [inline tomlrb.gemspec:21]: Why is this needed now and where is it used? Sorry I've been away from the ruby world for a while.

**Review by fbernier — APPROVED** [review]
Looks good other than my question, thanks. Merge away whenever you want.

File: tomlrb.gemspec (hunk 1 above)
Comment (KitaitiMakoto) [reply tomlrb.gemspec:21]: Ruby 3.1 upgraded Psych from 3 to 4 that includes API change, therefore we need upgrade Psych to 4 even when we use Ruby 3.0 to keep code simple.

**Comment by KitaitiMakoto** [comment]
Thank you, I will merge this pull request.
